cassandra-mesos-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/executor/BackupManager.java

```diff
@@ -0,0 +1,127 @@
+
+        LOGGER.info("Clearing snapshot of keyspace {}", keyspace);
+        storageServiceProxy.clearSnapshot(snapshot, keyspace);
+    }
+
+    private void copyKeyspaceSnapshot(final String snapshot, final String keyspace) throws IOException {
+        final List<String> tables = jmxConnect.getColumnFamilyNames(keyspace);
+        for (final String table : tables)
+            copyTableSnapshot(snapshot, keyspace, table);
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    private void copyTableSnapshot(final String snapshot, final String keyspace, final String table) throws IOException {
+        final File srcDir = findTableSnapshotDir(keyspace, table, snapshot);
+        final File destDir = new File(backupDir, keyspace + "/" + table);
+        destDir.mkdirs();
+
+        final File[] files = srcDir.listFiles();
+        if (files != null) {
+            for (final File file : files) {
+                if (file.isFile()) {
+                    Files.copy(file.toPath(), new File(destDir, file.getName()).toPath(), StandardCopyOption.REPLACE_EXISTING);
+                }
+            }
+        }
+    }
+
+    private File findTableSnapshotDir(final String keyspace, final String table, final String snapshot) {
+        final File dataDir = new File(jmxConnect.getStorageServiceProxy().getAllDataFileLocations()[0]);
+        final File keyspaceDir = new File(dataDir, keyspace);
+
+        final File tableDir = findTableDir(keyspaceDir, table);
+        final File snapshotDir = new File(tableDir, "/snapshots/" + snapshot);
+        if (!snapshotDir.exists()) throw new IllegalStateException("Snapshot dir does not exist: " + snapshotDir);
+
+        return snapshotDir;
+    }
+
+    private File findTableDir(final File keyspaceDir, final String table) {
+        final File[] files = keyspaceDir.listFiles();
+        if (files != null)
+            for (final File file : files) {
+                if (file.isDirectory() && file.getName().startsWith(table + "-")) {
+                    return file;
+                }
+            }
+
+        throw new IllegalStateException("Failed to found table dir for table " + table + " in keyspace " + keyspaceDir);
+    }
+
+    public void restore(@NotNull final String keyspace, final boolean truncateTables) throws IOException, TimeoutException {
+        final List<String> tables = jmxConnect.getColumnFamilyNames(keyspace);
+
+        for (final String table : tables) {
+            if (truncateTables) {
+                LOGGER.info("Truncating {}/{}", keyspace, table);
+                jmxConnect.getStorageServiceProxy().truncate(keyspace, table);
+            }
+
+            LOGGER.info("Restoring backup of {}/{}", keyspace, table);
+            restoreTableSnapshot(keyspace, table);
+
+            LOGGER.info("Reloading SSTables for {}/{}", keyspace, table);
+            jmxConnect.getColumnFamilyStoreProxy(keyspace, table).loadNewSSTables();
+        }
+    }
+
```

Comment:
Same as my response above:

```Hm, I think partial failure of the job is okay so long as we do not continue trying to backup the keyspace and move on to the next one. Is my understanding correct there? Also, if there is an exception I think we should catch it immediately and partially fail the job there instead of suppressing the warning. We should catch the error sooner rather than later if possible.```